### ISSUE TITLE
Bug: Sharding the request list

### ISSUE THREAD

**Issue by SoujanyaPonnapalli** [RainBlock/verifier#5]
The verifier currently shards the storage updates incorrectly.

For executing the transaction in the sample/txn.bin which updates the following two accounts:
```
<Buffer 00 00 00 00 00 00 00 00 00 00 00 00 a1 e4 38 0a 3b 1f 74 96 73 e2 70 22 99 93 ee 55 f3 56 63 b4> and
<Buffer 00 00 00 00 00 00 00 00 00 00 00 00 5d f9 b8 79 91 26 2f 6b a4 71 f0 97 58 cd e1 c0 fc 1d e7 34>
```

The hashed address are:
```
<Buffer 51 70 55 3c 37 8d 94 cf 46 92 a1 63 d1 0b 9b e7 40 1b b8 0c e1 89 bc 4c c4 f7 a9 bd 1d 45 bd e3>
<Buffer b7 14 7c 94 61 44 fe dc de 6f 4c a7 1a 9d 41 24 64 05 63 54 28 19 25 22 fe a7 db 60 02 36 a5 02>
```

So the shards that should get the update requests are shard 5 and shard 11.
But, the shards that get the updates are 14 and 10 respectively.

**Comment by no2chem** [comment]
Sorry, but I don't think your hashed addresses are correct?
A simple test:
```javascript
const hash = require('bigint-hash')
hash.hashAsBuffer(hash.HashType.KECCAK256, Buffer.from('a1e4380a3b1f749673e270229993ee55f35663b4', 'hex'))
> <Buffer ef 6e bc 78 d7 3b 85 dc 7c 81 af 5c de 0d 18 a7 a6 f2 6a 6e cb 3b 74 c5 89 2e 8d c0 52 e8 4d 77>
```

**Comment by no2chem** [comment]
It would appear though, that the address provided by the verifier is 32 bytes.
Rather than have the storage node truncate it, I've pushed a fix to have the verifier correctly send only 20 bytes instead of 32.

**Comment by SoujanyaPonnapalli** [comment]
Yeah just ended up cross checking the shard computation, and understanding that the addresses are longer than 20 bytes. Sorry for the wrong description!

**Comment by no2chem** [comment]
closed by 01936a1
